include/deal.II/grid/grid_generator.h

```diff
@@ -547,6 +547,103 @@ namespace GridGenerator
+   * or CylindricalManifold (in 3d) with manifold id $0$, a
+   * TransfiniteInterpolationManifold with manifold id $1$, and a FlatManifold
+   * everywhere else. For more information on this topic see
+   * @ref GlossManifoldIndicator "the glossary entry on manifold indicators".
+   * The
+   * cell faces on the cylinder and surrounding shells have manifold ids of
+   * $0$, while the cell volumes adjacent to the shells (or, if they do not
+   * exist, the cylinder) have a manifold id of $1$. Put another way: this
+   * grid uses TransfiniteInterpolationManifold to smoothly transition from
+   * the shells (generated with GridGenerator::concentric_hyper_shells) to the
+   * bulk region. All other cell volumes and faces have manifold id
+   * numbers::flat_manifold_id and use FlatManifold. All cells with id
+   * numbers::flat_manifold_id are rectangular prisms aligned with the
+   * coordinate axes.
+   *
+   *
+   * @param tria Triangulation to be created. Must be empty upon calling this
+   * function.
+   *
+   * @param half_height The half height of the channel (y- to 0 or 0 to y+).
+   *
+   * @param length_pre The length of the channel from the left side (x-) to the center of the cylinder (0)
+   *
+   * @param length_post The length of the channel from the cylinder (0) to the right side (x+)
+   *
+   * @param shell_region_radius Radius of the layer of shells around the cylinder.
+   * This value should be between larger than 0.5 (the radius of the cylinder)
+   * and smaller than 1 (the half-length of the box around the cylinder).
+   *
+   * @param n_shells Number of shells to use in the shell layer.
+   *
+   * @param depth The depth of the simulation domaine (in 3D, the z axis)
+   *
+   * @param depth_division The number of division along the z axis
+   *
+   * @param skewness Parameter controlling how close the shells are
+   * to the cylinder: see the mathematical definition given in
+   * GridGenerator::concentric_hyper_shells.
+   *
+   * @param use_transfinite_region If `true`, then a tranfinite manifold is used
+   * in the intermediary region between the cylindrical hypershell and the
+   * channel.
+   *
+   * @param colorize If `true`, then assign different boundary ids to
+   * different parts of the boundary. For more
+   * information on boundary indicators see
+   * @ref GlossBoundaryIndicator "this glossary entry".
+   * The left boundary (at $x = -L_{pre}$) is assigned an id of $0$, the right
+   * boundary (at $x = L_{post}$) is assigned an id of $1$; the boundary of
+   * the obstacle in the middle (i.e., the circle in 2d or the cylinder
+   * walls in 3d) is assigned an id of $2$, the bottom wall (at $y=-H$) is
+   * assigned and id of $/$, the top wall (at $y=H$) is assigned an id of $4$.
+   * In 3D, the front wall ($z=0$) is assigned an id of $5$ and the back wall
+   * ($z=W$) is assigned an id of $6$.
+   */
+  template <int dim>
+  void
+  custom_channel_with_cylinder(Triangulation<dim> &tria,
```

Comment:
We need to proceed with care since we now have three functions which do very similar things (plate_with_a_hole, channel_with_cylinder, and this one). The thing which makes this significantly different is that all cells (outside the cylinder) are cubes and I think the name should reflect that to distinguish it from the other two. How about `uniform_channel_with_cylinder()`? I don't know if we have a specific piece of jargon for "all cells are cubes" but that at least shows that things are identical, in some sense.

Otherwise - if reasonably possible I'd like to break up the parameter list with some `std::vector`s to improve legibility. I think that should be sufficient for this function.

Reply:
Awesome, I agree with this. I will proceed in this direction.
The cells are square in 2D, but don't have to be cubed in 3D. Still, uniform is a good prefix.